WPMedia/arc-themes-components
# THEMES 774 Additional icons to be added to the Icon component

## Description

In a review of [TMEDIA-788](https://arcpublishing.atlassian.net/browse/TMEDIA-788), design noticed that there are icons in [Figma](https://www.figma.com/file/giQllWgtzGk42KwK2XD9rY/Brand-Browsing-Page-Designs?type=design&node-id=2944-34134&mode=design&t=KJxzSikwWUdJorCs-0) that aren't available in the Icon component. My job in this ticket was to add the missing icons.
  
I made the following changes:
* Each icon got a file in the `icon/icons` directory. The file contains a `path` with a path for the icon's SVG
    * Most of the icons were sourced from [Figma](https://www.figma.com/file/giQllWgtzGk42KwK2XD9rY/Brand-Browsing-Page-Designs?type=design&node-id=2944-34134&mode=design&t=KJxzSikwWUdJorCs-0), but some were sourced from the [Storybook](https://5eed0506faad4f0022fedf95-nrlvhjyiiz.chromatic.com/?path=/docs/icon--all-icons-resize) because of Figma's weird handling of SVGs. Some of the icons would have incorrect filling or be missing when they were imported directly from Figma (this held for existing icons too during my testing)
    * To resize the icons' paths for our 512 x 512 view box, I used [this website](https://aydos.com/svgedit/)
* In `icon/icons/index.js` I added lines to export all of the new icons
    * **Icon names are taken from their names in Figma**
* In `icon/index.jsx` I added all of the new icons to the `name` prop, so they can be selected in PageBuilder
* I created a test block in `arc-themes-feature-pack` to display and verify the look of the new icons (see test steps below)

## Ticket

[THEMES-774](https://arcpublishing.atlassian.net/browse/THEMES-774)

## Acceptance Criteria

1. Missing icons are added to the Icon component repository

## Test Steps

1. Checkout this branch: `git checkout THEMES-774-additional-icons`
2. Update dependencies: `npm i`
3. Run Storybook: `npm run storybook`
4. Check out the Icon storybook documentation (**"All Icons" page**)
    * Make sure all of the icons referenced in [Figma](https://www.figma.com/file/giQllWgtzGk42KwK2XD9rY/Brand-Browsing-Page-Designs?type=design&node-id=2944-34134&mode=design&t=KJxzSikwWUdJorCs-0) are present
    * Make sure the icons look good (no filling issues or obvious flaws)
5. **If you want to see the new icons in PageBuilder,** follow the steps below:
    * Go to `arc-themes-feature-pack`
    * Checkout this branch: `git checkout THEMES-774-additional-icons` (**This branch will be deleted once this ticket/branch gets merged in!**)
    * Start fusion (no blocks need to be linked) with `npx fusion start`
    * Open your local PageBuilder instance and create a new page
        * Select the Right Rail layout and insert an "icon-test default" bock in the "Main" section
    * All of the new icons will display on the page. Verify that they look good and that there are no issues with filling, sizing, etc.

## Effect of Changes

### Before

The icons below were not available in PageBuilder:

![Screenshot 2023-08-14 at 1 16 15 PM](https://github.com/WPMedia/arc-themes-components/assets/103296586/fce94d4b-da17-48f4-b386-b4f23f947809)

### After

The icons from above are now usable in PageBuilder (ignore the color, it was for testing):

![Screenshot 2023-08-14 at 1 17 22 PM](https://github.com/WPMedia/arc-themes-components/assets/103296586/45cc296d-407c-43e0-ac3d-982f8fa5973f) ![Screenshot 2023-08-14 at 1 18 03 PM](https://github.com/WPMedia/arc-themes-components/assets/103296586/cd2f8f25-1db8-42bc-9768-890f955cfdc3) ![Screenshot 2023-08-14 at 1 24 07 PM](https://github.com/WPMedia/arc-themes-components/assets/103296586/a7c023ed-c3f0-4797-8d96-fd067161690e)

## Author Checklist


## Reviewer Checklist

_The reviewer of the PR should copy-paste this template into the review comments on review._



[TMEDIA-788]: https://arcpublishing.atlassian.net/browse/TMEDIA-788?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[THEMES-774]: https://arcpublishing.atlassian.net/browse/THEMES-774?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
